### PR TITLE
Add an illegal file name error type

### PR DESCRIPTION
**Pull request by MrCreosote**

### CODE REVIEW

File: docs/import_specifications.ADR.md [hunk 1]
```diff
@@ -148,14 +148,14 @@ each error type. Currently the error types are:
...
+* `illegal_file_name` if an illegal file name is provided
 * `multiple_specifications_for_data_type` if more than one tab or file per data type is submitted
 * `no_files_provided` if no files were provided
 * `unexpected_error` if some other error occurs
 
 The HTTP code returned will be, in order of precedence:
 
-* 400 if at least one error is `cannot_parse_file`, `no_files_provided`, or
-  `multiple_specifications_for_data_type`
+* 400 if any error other than `cannnot_find_file` or `unexpected_error` occurs
```
Comment (briehl) [inline docs/import_specifications.ADR.md:158]: Typo - an extra `n`

File: docs/import_specifications.ADR.md [hunk 2]
```diff
@@ -148,14 +148,14 @@ each error type. Currently the error types are:
 
 * `cannot_find_file` if an input file cannot be found
 * `cannot_parse_file` if an input file cannot be parsed
+* `illegal_file_name` if an illegal file name is provided
```
Comment (briehl) [inline docs/import_specifications.ADR.md:151]: What makes the file name illegal?

File: docs/import_specifications.ADR.md [hunk 3]
```diff
@@ -196,6 +196,18 @@ entries, but will not do further error checking.
...
+#### `illegal_file_name`
+
+```
+{"type": "illegal_file_name",
+ "message": <message regarding the illegal name>,
+ "file": <filepath>
+}
+```
+
+The `file` key will not be provided if there is a `null` filename provided. This is currently
```
Comment (briehl) [inline docs/import_specifications.ADR.md:208]: 👍

**Review by briehl — CHANGES_REQUESTED** [review]
Just a question and a typo.

File: docs/import_specifications.ADR.md (hunk 2 above)
Comment (MrCreosote) [reply docs/import_specifications.ADR.md:151]: I'm thinking of the case like `?files=file1.csv,,file2.csv,   ,file3.csv`

Where `""` is translated to `"."` by `pathlib` and the whitespace only files, while officially supported by posix, are not something we should support.

File: docs/import_specifications.ADR.md (hunk 1 above)
Comment (MrCreosote) [reply docs/import_specifications.ADR.md]: fixed

File: docs/import_specifications.ADR.md (hunk 2 above)
Comment (briehl) [reply docs/import_specifications.ADR.md:151]: Got it.

**Review by briehl — APPROVED** [review]

**Pull request merged**